README.md

```diff
@@ -11,6 +11,135 @@ supported boards and instructions on how to setup environment and build images.
+meta-mender-community is a top level layer which is not actually a layer it
+self but contains a collection of layers which integrates Mender on various
+boards.
+
+The layers are structured based on upstream BSP layers and not individual
+boards.
+
+There are several types of BSP layers, some are SoC oriented and a selection of
+those would be:
+
+- meta-freescale
+- meta-sunxi
+- meta-rockchip
+- meta-renesas
+
+And then you also have System on Module vendors that typically have their own
+BSP layers which append to the SoC ones, and a selection of those would be:
+
+- meta-variscite
+- meta-phytec
+- meta-toradex-nxp
+
+Naming of integration layers in meta-mender-community follows the upstream
+naming closely where SoM vendor layer name take precedence and the convention
+is:
+
+   meta-mender-<upstream suffix>
+
+This should make it clear which layer is targeting what BSP.
+
+### Manifest files
+
+We utilize the [google repo](https://source.android.com/setup/develop/repo) tool
+and the usage of manifest files (xml). There is a common manifest for all
+integration layers that can be found at:
+
+    meta-mender-community/scripts/mender.xml
+
+This includes the required Mender layers,
+[meta-mender](https://github.com/mendersoftware/meta-mender) and
+[meta-mender-community](https://github.com/mendersoftware/meta-mender-community).
+
+Each integration layer should provide a manifest file, e.g
+
+    meta-mender-community/meta-mender-sunxi/scripts/manifest-sunxi.xml
+
+Above includes `mender.xml`
+
+### Templates and configuration fragments
+
+The typical work flow utilizing this layer is,
+
+download the source:
+
+    $ mkdir mender-<vendor/soc name>
+    $ cd mender-<vendor/soc name>
+    $ repo init \
+           -u https://github.com/mendersoftware/meta-mender-community \
+           -m meta-mender-<vendor/soc name>/scripts/manifest-<vendor/soc name>.xml \
+           -b sumo
+
+setup environment
+
+    $ . setup-environment <vendor/soc name>
+
+build
+
+    $ bitbake core-image-base
+
+The magic here is the `setup-environment` script which is a wrapper for
+`oe-init-build-env` which you typically call to setup the Bitbake environment.
+
+What `setup-environment` does is:
+
+- source `oe-init-build-env`
+- append common Mender changes to `local.conf`
+    - content of `meta-mender-community/templates/local.conf.append`
+- append integration layer specific changes to `local.conf`
+    - content of `meta-mender-community/meta-mender-<vendor/soc name>/templates/local.conf.append`
+- copy `bblayers.conf` template
+    - `meta-mender-community/meta-mender-<vendor/soc name>/templates/bblayers.conf.sample`
+
+Meaning that each integration layer must provide:
+
+- an `local.conf.append` file containing any board specific changes that are
+required for Mender integration.
+- `bblayers.conf.sample` which is a template that will be copied to
+`bblayers.conf`
+
+### Integration layer examples
+
+Lets start with taking a closer look at meta-mender-qemu:
```

Comment:
Is this (all the way to the end) valuable? I think it can be dropped from the readme.

Reply:
I think the whole section "Integration Layer Examples" can be removed.